skbio/io/format/stockholm.py

```diff
@@ -651,6 +700,21 @@ def _tabular_msa_to_stockholm(obj, fh):
             for tree_id, tree in obj.metadata[gf_feature].items():
                 fh.write("#=GF TN %s\n" % tree_id)
                 fh.write("#=GF NH %s\n" % tree)
+        elif gf_feature == 'RN':
+            if not isinstance(gf_feature_data, list):
+                raise StockholmFormatError("Expected 'RN' to contain a list "
+                                           "of dictionaries, got %s."
+                                           % gf_feature_data)
+            for ref_num, dictionary in enumerate(gf_feature_data, start=1):
```

Comment:
Can you type-check that `dictionary` is a `dict` instance at the beginning of the loop, similar to the type-checking above for `gf_feature_data`?